src/csharp/Microsoft.Spark/Sql/Types/ComplexTypes.cs

```diff
@@ -51,21 +51,23 @@ public ArrayType(DataType elementType, bool containsNull = true)
         /// <summary>
         /// Returns JSON object describing this type.
         /// </summary>
-        internal override object JsonValue =>
-            new JObject(
-                new JProperty("type", TypeName),
-                new JProperty("elementType", ElementType.JsonValue),
-                new JProperty("containsNull", ContainsNull));
+        internal override object JsonValue => JsonDocument.Parse(JsonSerializer.Serialize(
```

Comment:
For perf, consider using the `Utf8JsonWriter` rather than `JsonSerializer.Serialize` to create the JSON before passing it to `JsonDocument.Parse`.